examples/idx_bd_cache/main.go

```diff
@@ -15,145 +15,216 @@
 package main
 
 import (
-	"fmt"
+	"time"
+
 	"github.com/ligato/cn-infra/core"
 	"github.com/ligato/cn-infra/datasync"
-	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/datasync/kvdbsync"
+	"github.com/ligato/cn-infra/flavors/localdeps"
 	log "github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/cn-infra/utils/safeclose"
 	"github.com/ligato/vpp-agent/flavors/vpp"
 	"github.com/ligato/vpp-agent/plugins/defaultplugins"
 	"github.com/ligato/vpp-agent/plugins/defaultplugins/l2plugin/bdidx"
 	"github.com/ligato/vpp-agent/plugins/defaultplugins/l2plugin/model/l2"
 	"github.com/ligato/vpp-agent/plugins/defaultplugins/l2plugin/testing"
-	"time"
 )
 
 // Start Agent plugins selected for this example
 func main() {
 	// Init close channel to stop the example
-	closeChannel := make(chan struct{}, 1)
+	exampleFinished := make(chan struct{}, 1)
 
-	f := vpp.Flavor{}
-	// Example plugin will show index mapping
-	examplePlugin := &core.NamedPlugin{PluginName: PluginID, Plugin: &examplePlugin{
-		agent1: f.ETCDDataSync.OfDifferentAgent("agent1", f),
-		agent2: f.ETCDDataSync.OfDifferentAgent("agent2", f)}}
+	// Start Agent with ExampleFlavor (combinatioplugin.GoVppmux, n of ExamplePlugin & reused cn-infra plugins)
+	flavor := ExampleFlavor{closeChan: &exampleFinished}
+	agent := core.NewAgent(log.DefaultLogger(), 15*time.Second, append(flavor.Plugins())...)
+	core.EventLoopWithInterrupt(agent, exampleFinished)
+}
 
-	// Create new agent
-	agent := core.NewAgent(log.DefaultLogger(), 15*time.Second, append(f.Plugins(), examplePlugin)...)
+/**********
+ * Flavor *
+ **********/
+
+// ExampleFlavor is a set of plugins required for the datasync example.
+type ExampleFlavor struct {
+	// Local flavor to access to Infra (logger, service label, status check)
+	*vpp.Flavor
+	// Example plugin
+	IdxBdCacheExample ExamplePlugin
+	// For example purposes, use channel when the example is finished
+	closeChan *chan struct{}
+}
+
+// Inject sets object references
+func (ef *ExampleFlavor) Inject() (allReadyInjected bool) {
+	// Init local flavor
+	if ef.Flavor == nil {
+		ef.Flavor = &vpp.Flavor{}
+	}
+	ef.Flavor.Inject()
 
-	// End when the idx_bd_cache example is finished
-	go closeExample("idx_bd_cache example finished", closeChannel)
+	// Inject infra + transport (publisher, watcher) to example plugin
+	ef.IdxBdCacheExample.PluginInfraDeps = *ef.FlavorLocal.InfraDeps("idx-bd-cache-example")
+	ef.IdxBdCacheExample.Publisher = &ef.ETCDDataSync
+	ef.IdxBdCacheExample.Agent1 = ef.ETCDDataSync.OfDifferentAgent("agent1", ef)
```

Comment:
Please try to use: 

ef.Flavor.FlavorETCD.ETCDDataSync to overcome following golint message:

call of f.ETCDDataSync.OfDifferentAgent copies lock value: github.com/ligato/vpp-agent/flavors/vpp.Flavor contains github.com/ligato/vpp-agent/plugins/govppmux.GOVPPPlugin contains sync.WaitGroup contains sync.noCopy

Please, check also if build fails on this error (it should, but it probably fails not).